audio-debrief-recording/src/pages/home/home.ts

```diff
@@ -0,0 +1,129 @@
+import { Component } from '@angular/core';
+import { NavController } from 'ionic-angular';
+import { MediaObject } from '@ionic-native/media';
+import { File, DirectoryEntry, FileEntry } from '@ionic-native/file';
+
+@Component({
+  selector: 'page-home',
+  templateUrl: 'home.html'
+})
+export class HomePage {
+
+  audio: MediaObject;
+  playing = false;
+  buttonDisabled = true;
+  fileName = 'audio_debrief.m4a';
+  fileLength = '';
+  fileSize = '';
+
+  constructor(public navCtrl: NavController, private file: File) {
+  }
+
+  recordButtonText = 'Record audio';
+  startingRecording = true;
+  playButtonText = 'Play';
+
+  recordAudio() {
+    if (this.startingRecording) {
+      this.file.createFile(this.file.dataDirectory, this.fileName, true).then(() => {
+        let src = this.file.dataDirectory.replace(/^file:\/\//, '') + this.fileName;
+        // ionic media.create is buggy (getDuration returns always -1), the solution is to use raw cordova API
+        this.audio = new (<any>window).Media(src, this.mediaSuccess, this.mediaError, this.mediaStatus);
```

Comment:
any type file for this?

Reply:
we had to use here Cordova API and unfortunately, there are no TS typings for it. (Ionic wrapper wasn't working here - getDuration() method was not working)
If you check the **this.audio** type you will see that actually we are using Ionic media type (as it has the same API but is not the same 'class')